### PR TITLE
Remove k8s lookup, use k8s_facts which does not require manual env var setting

### PR DESCRIPTION
**Pull request by djwhatle**
# Overview
Moves from using the k8s lookup plugin to the k8s_facts module.

This change removes the dependency on whatever the user has set for the `KUBECONFIG` environment variable before running the playbook. With k8s_facts, the playbook will _always_ use the values in `auth/kubeconfig` (since we set them in the environment https://github.com/fusor/ocp-velero-ansible/blob/master/create-nginx.yml#L4-L5) created by openshift-installer. 

# Why
Before this PR, the playbook outputs a kubeconfig path that will be "used" for the duration of the playbook...
```
TASK [login_ocp : debug] ****************************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "Logging into cluster using KUBECONFIG: /home/dwhatley/dev/ocp-velero-ansible/auth/kubeconfig"
}
```
... but in reality all calls to lookup plugins (including the k8s lookup plugin) will use whatever `KUBECONFIG` env var you have set in your local environment. This leads to a confusing error condition where some tasks will run successfully but others won't if you have a bad KUBECONFIG set in your env.

Below is an example of what you might see on a task that uses the lookup plugin if your `KUBECONFIG` env var wasn't set properly before running the playbook.
```
TASK [create_nginx_app_with_backup : Wait 2 minutes for nginx deployment] ********************************************************************************************************************
2019-02-15 13:20:32,652 WARNING Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7fb05437d350>: Failed to establish a new connection: [Errno 113] No route to host',)': /version
```

## Notes / Testing Needs
I haven't been able to execute a successful backup on my workstation yet, so both of the changed roles still need testing before merge. Hopefully whoever has a working environment can give them a shot.

### CODE REVIEW

**Comment by djwhatle** [comment]
Got my cluster to a good state where I could do some testing. Found an issue with namespace deletion step.

```
TASK [delete_nginx_and_restore : Wait 2 minutes for namespace to be deleted] ************************************************************************************************************************************
FAILED - RETRYING: Wait 2 minutes for namespace to be deleted (10 retries left).
FAILED - RETRYING: Wait 2 minutes for namespace to be deleted (9 retries left).
FAILED - RETRYING: Wait 2 minutes for namespace to be deleted (8 retries left).
FAILED - RETRYING: Wait 2 minutes for namespace to be deleted (7 retries left).
FAILED - RETRYING: Wait 2 minutes for namespace to be deleted (6 retries left).
FAILED - RETRYING: Wait 2 minutes for namespace to be deleted (5 retries left).
FAILED - RETRYING: Wait 2 minutes for namespace to be deleted (4 retries left).
FAILED - RETRYING: Wait 2 minutes for namespace to be deleted (3 retries left).
FAILED - RETRYING: Wait 2 minutes for namespace to be deleted (2 retries left).
FAILED - RETRYING: Wait 2 minutes for namespace to be deleted (1 retries left).
fatal: [localhost]: FAILED! => {"attempts": 10, "changed": false, "resources": []}
```

**Comment by djwhatle** [comment]
I believe I've also fixed https://github.com/fusor/ocp-velero-ansible/issues/5 by chaining "get()" calls which allow for safely delving into a nested dict structure.

https://github.com/fusor/ocp-velero-ansible/pull/7/files#diff-d5002c305caa367401126e6a483d9492R32

@dymurray I've tested this, and it's ready to review / merge.

File: roles/create_nginx_app_with_backup/tasks/main.yml [hunk 1]
```diff
@@ -3,34 +3,33 @@
...
+  k8s_facts:
+    api_version: "apps/v1"
+    kind: "Deployment"
+    namespace: "nginx-example"
+    label_selectors:
+      - "app = nginx"
+  register: deploy
+  until: deploy
+         and (deploy.resources | length > 0)
+         and (deploy.resources[0].get("status", {}).get("replicas",-1) == deploy.resources[0].get("status", {}).get("availableReplicas", -2))
```
Comment (dymurray) [inline roles/create_nginx_app_with_backup/tasks/main.yml:17]: Just curious, what is the second argument to `get()`? Why use `-1` and `-2`?

**Review by dymurray — APPROVED** [review]
Looks good, just curious on how the getter works

File: roles/create_nginx_app_with_backup/tasks/main.yml (hunk 1 above)
Comment (dymurray) [reply roles/create_nginx_app_with_backup/tasks/main.yml:17]: Just kidding saw your comments in slack. It's the python `get` which allows us to set a default. Looks great!

**Pull request merged**